packages/safe-tools-client/app/components/network-connect-modal/index.gts

```diff
@@ -0,0 +1,197 @@
+    if (this.isConnected) {
+      this.args.onConnect?.();
+    }
+  }
+
+  get radioWalletProviderId() {
+    // TODO
+    return this.walletProviders[0].id;
+  }
+
+  @action cancelConnection() {
+    // TODO
+  }
+
+  <template>
+    <BoxelModal
+      class='network-connect-modal'
+      @size='medium'
+      @isOpen={{@isOpen}}
+      @onClose={{@onClose}}
+      data-test-network-connect-modal={{@name}}
+    >
+      <div class='network-connect-modal__scroll-wrapper'>
+        <BoxelActionContainer
+          class='network-connect-modal__card'
+          tabindex='-1'
+          {{focusTrap
+            isActive=@isOpen
+            focusTrapOptions=(hash
+              allowOutsideClick=true
+              clickOutsideDeactivates=true
+              initialFocus='.network-connect-modal__card'
+            )
+          }}
+          {{!
+            Focus is trapped within this element, with the element as the initial focus
+            outside clicks are allowed to trigger @onClose
+
+            This should ideally be replaced with a selector for the primary/first action
+            in the modal. However, the programmatic focus it introduces does not match
+            :focus-visible. This behaviour means the user doesn't know what is being focused.
+            Forcing the user to press Tab will create focus that does match :focus-visible
+            and hence show a ring.
+
+            Also, there is no distinguishing state for focused radio buttons - leading to the same
+            problem of the user not knowing what is being focused, if we try to select the radio
+            button for layer one as the initial focus.
+
+            Some reading, if you are up for a slight headache:
+            - https://github.com/WICG/focus-visible/issues/88
+            - https://github.com/w3c/csswg-drafts/issues/5885
+          }}
+
+          as |Section ActionChin|
+        >
+          <button
+            class='network-connect-modal__close-button'
+            type='button'
+            aria-label='Close'
+            {{on 'click' @onClose}}
+          >
+            {{svgJar 'close' width='100%' height='100%' aria-hidden=true}}
+          </button>
+
+          <Section @title="Connect your wallet">
+            <BoxelRadioInput
+              @name="wallet-provider"
+              @orientation="horizontal"
+              @spacing="default"
+              @groupDescription='Select a wallet to connect to'
+              @items={{this.walletProviders}}
+              @disabled={{@isConnecting}}
+              @checkedId={{@currentWalletProviderId}}
+              @hideRadio={{true}}
+              class='network-connect-modal__wallet-group'
+              data-test-wallet-selection as |option|
+            >
+              {{#let option.data as |item|}}
+                <option.component
+                  @name='wallet-provider-selection'
+                  @onChange={{fn (optional @changeWalletProvider) item.id}}
```

Comment:
We should remember to remove the `optional` later... though with this implementation, it seems possible to add logic for updating the selected item within this component, since it is a local radio selection that only affects external/global state when "Connect Wallet" is pressed.

Reply:
👍🏻 I’ve made this an internal property now